src/LOAuth.ts

```diff
@@ -112,6 +112,67 @@ class LOAuth {
     return btoa(JSON.stringify(state));
   }
 
+  /**
+   * Sets a domain cookie for the token data in a format compatible with this library
+   * The cookie can be read by another application on a compatible domain by triggering
+   * refreshAuthToken, which will find and import the cookie as a token
+   */
+  public setDomainCookieAuthState(token: LOAuth.CookieTokenData) {
+    document.cookie = `lo-app-tools-auth=${JSON.stringify({
+      access_token: token.access_token,
+      refresh_token: token.refresh_token,
+      expires: token.expires,
+      clientId: token.clientId,
+      cookieDomain: token.cookieDomain
+    })};domain=.${token.cookieDomain};Max-Age=10;path=/;secure`;
+  }
+
+  /**
+   * This method looks for a cookie set by setDomainCookieAuthState that allows the
+   * applications to exchange auth tokens over different subdomains
+   */
+  private _getDomainCookieAuthState() {
+    try {
+      if (!document.cookie) {
+        return;
+      }
+      const cookie = document.cookie
+        .split('; ')
+        .find((cookie) =>
+          cookie.startsWith(
+            `${this.clientOptions.storageKey || AUTH_STORAGE_KEY}=`
+          )
+        );
+
+      const value = cookie && cookie.split('=')[1];
+      if (!value) {
+        return;
+      }
+
+      const storedData: LOAuth.CookieTokenData = JSON.parse(value);
+      const token = this.client.createToken({
+        access_token: storedData.access_token,
+        refresh_token: storedData.refresh_token,
+        token_type: 'Bearer'
+      });
+      token.expiresIn(storedData.expires - Date.now());
+
+      // the cookie clientId takes precedent over configured options
+      this.clientOptions.clientId = storedData.clientId;
+
+      // delete the cookie after it's read
+      document.cookie = `${AUTH_STORAGE_KEY}=;domain=.${storedData.cookieDomain};Max-Age=-9999;path=/;secure`;
```

Comment:
`Max-Age=-9999` is what does the delete trick here right? Might be worth commenting that

Reply:
👍🏽 will add in the follow up